web/modules/custom/joinup_federation/tests/src/Kernel/UserSelectionFilterWhitelistBuilderTest.php

```diff
@@ -1,173 +0,0 @@
-<?php
```

Comment:
To be honest, the main reason is that it was too hard for me to write another one and compared to the effort needed, I didn't think it was worth more than the behat test.
The previous test was based on the UserSelectionFilter step and specifically the `::buildWhitelist`. The `::buildWhitelist` method was a recursive method running after a user was submitting the form, in a single run (not batched) and was calculating the entities whitelisted. The method was also not interacting with the pipeline or the batch storage (`::set/getBatchValue`) while building the list.

The execution method of the analyze step is mainly doing all of these.. That makes it hard to run standalone, without preparing a pipeline or in a UnitTest (this last one is actually a nightmare to me since I am not familiar with it).

On the other side, the new behat test that was extended, is checking trice the list of entities as shown in the UserSelectionFilter step, so after they are analyzed. All cases are checked fine with more complete data and run through the whole process of the importer.

On the bottom line, we were maintaining a list of custom solutions in the test, a duplicated array, that was supposed to check exactly what we see in the behat test as well. We do not assert that the hashes are the same of course, but we do assert the end result.

Still, the bottom line is that I initially tried, but all the dependencies, the pipeline, the previous step, the batch processing, they were quite a lot of effort for something we already test in behat.

Reply:
I added a test.